core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup.go

```diff
@@ -0,0 +1,432 @@
+	q := url.Values{
+		FeedID:    {strings.Join(ml.feeds, ",")},
+		Timestamp: {ml.time.String()},
+		UserId:    {upkeepId.String()},
+	}
+
+	reqUrl := fmt.Sprintf("%s%s%s", r.mercury.cred.URL, MercuryBatchPathV3, q.Encode())
+	r.lggr.Debugf("FeedLookup request URL: %s", reqUrl)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, reqUrl, nil)
+	if err != nil {
+		ch <- MercuryBytes{Index: 0, Error: err}
+		return
+	}
+
+	ts := time.Now().UTC().UnixMilli()
+	signature := r.generateHMAC(http.MethodGet, MercuryBatchPathV3+q.Encode(), []byte{}, r.mercury.cred.Username, r.mercury.cred.Password, ts)
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", r.mercury.cred.Username)
+	req.Header.Set("X-Authorization-Timestamp", strconv.FormatInt(ts, 10))
+	req.Header.Set("X-Authorization-Signature-SHA256", signature)
+
+	retryable := false
+	retryErr := retry.Do(
+		func() error {
+			resp, err1 := r.hc.Do(req)
+			if err1 != nil {
+				r.lggr.Errorf("FeedLookup upkeep %s block %s GET request fails for multi feed: %v", upkeepId.String(), ml.time.String(), err1)
+				return err1
+			}
+			defer resp.Body.Close()
+			body, err1 := io.ReadAll(resp.Body)
+			if err1 != nil {
+				return err1
+			}
+
+			if resp.StatusCode == http.StatusNotFound || resp.StatusCode == http.StatusInternalServerError {
+				r.lggr.Errorf("FeedLookup upkeep %s block %s received status code %d for multi feed", upkeepId.String(), ml.time.String(), resp.StatusCode)
+				retryable = true
+				return errors.New(strconv.FormatInt(int64(resp.StatusCode), 10))
+			} else if resp.StatusCode != http.StatusOK {
+				return fmt.Errorf("FeedLookup upkeep %s block %s received status code %d for multi feed", upkeepId.String(), ml.time.String(), resp.StatusCode)
+			}
+
+			var m MercuryResponse
+			err1 = json.Unmarshal(body, &m)
+			if err1 != nil {
+				r.lggr.Errorf("FeedLookup upkeep %s block %s failed to unmarshal body to MercuryResponse for multi feed: %v", upkeepId.String(), ml.time.String(), err1)
+				return err1
+			}
+			blobBytes, err1 := hexutil.Decode(m.ChainlinkBlob)
+			if err1 != nil {
+				r.lggr.Errorf("FeedLookup upkeep %s block %s failed to decode chainlinkBlob %s for multi feed: %v", upkeepId.String(), ml.time.String(), m.ChainlinkBlob, err1)
+				return err1
+			}
+			ch <- MercuryBytes{
+				Index: 0,
+				Bytes: blobBytes,
+			}
+			return nil
+		},
+		// only retry when the error is 404 Not Found or 500 Internal Server Error
+		retry.RetryIf(func(err error) bool {
+			return err.Error() == fmt.Sprintf("%d", http.StatusNotFound) || err.Error() == fmt.Sprintf("%d", http.StatusInternalServerError)
+		}),
+		retry.Context(ctx),
+		retry.Delay(RetryDelay),
+		retry.Attempts(TotalAttempt))
+
+	// if all retries fail, return the error and ask the caller to handle cool down and heavyweight retry
+	if retryErr != nil {
+		mb := MercuryBytes{
+			Index:     0,
+			Retryable: retryable,
```

Comment:
Not sure how want to handle this, but what happens when we have a first request that is retryable, then we get a real error in the second try?
As I understand it would result in a retryable upkeep, as we don't set `retryable` to false before each try

Reply:
good catch. added logic and tests to cover this.